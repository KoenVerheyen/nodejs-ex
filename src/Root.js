// @flow
import React from 'react'
import {BrowserRouter, Switch, Route} from 'react-router-dom'
import {connect} from "react-redux";
import RadPageComponent from "./MainPage/RadPageComponent";

type Props = {
    connected: boolean
}

class Root extends React.Component<Props> {

    render() {
       return (
            <BrowserRouter>
                <Switch>
                    <Route path='/rad' component={RadPageComponent}/>
                    <Route render={(props) => <Page404 />}/>
                </Switch>
            </BrowserRouter>
        )
    }
}

function Page404() {
    return (<div>404</div>)
}

export default connect((state, ownProps) => {return {connected: state.rad.iam.connected}})(Root)