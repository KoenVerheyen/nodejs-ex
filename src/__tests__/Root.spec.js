import React from 'react'
import {shallow} from 'enzyme'
import configureMockStore from 'redux-mock-store'
import Root from '../Root'

const mockStore = configureMockStore([])

describe('rendering', () => {

    let oldenv

    beforeEach(() => {
        oldenv = process.env.NODE_ENV
    })

    afterEach(() => {
        process.env.NODE_ENV = oldenv
    })

    test('connected root', () => {
        let store = mockStore({rad: { iam: {connected: true}}})
        let wrapper = shallow(<Root store={store}/>).shallow()

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.length).toEqual(1)


    })


})