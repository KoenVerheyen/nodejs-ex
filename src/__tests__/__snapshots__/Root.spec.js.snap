// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering connected root 1`] = `
<Component>
  <Switch>
    <Route
      component={[Function]}
      path="/rad"
    />
    <Route
      render={[Function]}
    />
  </Switch>
</Component>
`;
